app/models/vm_or_template.rb

```diff
@@ -1881,10 +1881,16 @@ def self.includes_template?(ids)
 
   def self.cloneable?(ids)
     vms = VmOrTemplate.where(:id => ids)
-    return false if vms.blank?
+    return false if vms.blank? || vms.any? {|vm| vm.orphaned? || vm.archived? }
     vms.all?(&:cloneable?)
   end
 
+  def self.publishable?(ids)
```

Comment:
@romanblanco talking to @martinpovolny from backend point of view, we should start to unify actions on models, the new button implementation should expect unified interface.

So most of the actions now have, E.g. for action 'restart', there is method 'validate_restart' that is being called by is_available?(:restart)

So this should be really rewritten to the same way, so you call vms.all?{|x| x.is_available?(:clone)} # same for publish method

And if needed there can be class method self.is_available?(ids, feature)

I am